week-2-wide-perspective/w02e04-dungeons-and-the-princess/src/models/Castle.ts

```diff
@@ -0,0 +1,19 @@
+import { Person } from "./Person";
+import { Dungeon } from "./Dungeon";
+
+export class Castle {
+  constructor(private dungeon: Dungeon) {}
+
+  gotoTheDungeon(daredevil: Person) {
+    // #RQ1: peasant cannot be at the Castle !
+    try {
+      if (daredevil.title === "peasant") {
+        throw new Error(" peasant cannot be at the Castle !");
+      }
```

Comment:
To nie może tak być.

Znaczy się tak - `throw` i `Error` to dobry pomysł bardzo.
Ale to nie może być lokalnie złapane

Czyli nie możesz tutaj gdzie rzucasz błąd - od razu go łapać.

`try...catch` powinien być zrobiony w `main.ts` i to z poziomu `main.ts` musisz określić:
- co poszło nie tak
- z którego miejsca jest błąd.